crates/sage/src/scoring.rs

```diff
@@ -30,6 +31,7 @@ struct PreScore {
     peptide: PeptideIx,
     precursor_charge: u8,
     isotope_error: i8,
+    precursor_index: usize,
```

Comment:
This doubles the size of the struct (PeptideIx = 4 bytes + 1 byte for charge + 1 byte for isotope_error + 2 bytes for matched) by adding a `usize` (8 bytes) and thus doubles memory allocation in a hot loop.

If we can show it makes no difference in practice for open searches, then I suppose it's fine. We can probably steal a byte from `matched` (will we ever have > 255 matched peaks for a candidate?) and give it to `precursor_index` (will we ever have > 255 precursors for a spectrum?) or do some bitpacking across `precursor_charge` (given that this is normally in 0 - 8 range, e.g. 3 bits max)

Reply:
your concern was right, open search does become a lot slower ( 1m37.51s vs 2m15.60s in my laptop for a -100Da to +100Da search) ... looking into a more efficient way to represent this (I am thinking on moving the tracking of the precursor index to the InitialHits ...)

Reply:
New implementation does not store the precursor index info at the score level